src/corefx/System.Globalization.Native/collation.cpp

```diff
@@ -26,27 +32,213 @@ const int32_t CompareOptionsIgnoreCase = 1;
 typedef struct _sort_handle
 {
     UCollator* regular;
-    UCollator* ignoreCase;
+    TCollatorMap collatorsPerOption;
+    pthread_mutex_t collatorsLockObject;
 
-    _sort_handle() : regular(nullptr), ignoreCase(nullptr)
+    _sort_handle() : regular(nullptr)
     {
+        pthread_mutex_init(&collatorsLockObject, NULL);
```

Comment:
Does the mutex need to be cleaned up when we're done with it with pthread_mutex_destroy?